src/services/db/CommitServiceGithub.ts

```diff
@@ -59,8 +62,330 @@ export default class CommitServiceGitHub extends GitHubService {
 
     // We still await and throw error if the commit to staging-lite fails,
     // but the caller only gets the `resultToStaging` commit returned
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const [resultToStaging, _] = await Promise.all(createPromises)
 
     return resultToStaging
   }
+
+  async update(
+    sessionData: UserWithSiteSessionData,
+    {
+      fileContent,
+      sha,
+      fileName,
+      directoryName,
+    }: {
+      fileContent: string
+      sha: string
+      fileName: string
+      directoryName: string | undefined
+    }
+  ): Promise<GitCommitResult> {
+    const updatePromises = [
+      super.update(sessionData, {
+        fileContent,
+        sha,
+        fileName,
+        directoryName,
+        branchName: STAGING_BRANCH,
+      }),
+    ]
+    const shouldStagingLiteUpdate =
+      directoryName &&
+      !isFileAsset(directoryName) &&
+      isReduceBuildTimesWhitelistedRepo(sessionData.growthbook)
+
+    if (shouldStagingLiteUpdate) {
+      updatePromises.push(
+        super.update(sessionData, {
+          fileContent,
+          sha,
+          fileName,
+          directoryName,
+          branchName: STAGING_LITE_BRANCH,
+        })
+      )
+    }
+
+    // We still await and throw error if the commit to staging-lite fails,
+    // but the caller only gets the `resultToStaging` commit returned
+    const [resultToStaging, _] = await Promise.all(updatePromises)
+
+    return resultToStaging
+  }
+
+  async deleteDirectory(
+    sessionData: UserWithSiteSessionData,
+    {
+      directoryName,
+      message,
+      githubSessionData,
+      isStaging,
+    }: {
+      directoryName: string
+      message: string
+      githubSessionData: GithubSessionData
+      isStaging?: boolean
```

Comment:
Hmm why do we need to specify `isStaging`? Seems like it can be determined using `isFileAssets()`?

Similar comments to the other functions in this file.